packages/spec/helpers/test-helpers.js

```diff
@@ -13,60 +14,63 @@ const resolveFrom = require('resolve-from');
   const hopsBin = resolveFrom(cwd, 'hops/bin');
   const command = `${process.argv[0]} ${hopsBin} build ${argv.join(' ')}`;
+  const returnValue = { stdout: '', stderr: '' };
   debug('Starting', command);
   try {
-    return await exec(command, { env, cwd });
-  } catch (err) {
-    return { stderr: err.message };
+    const result = await exec(command, { env, cwd });
+    Object.assign(returnValue, result);
+  } catch (error) {
+    Object.assign(returnValue, { error });
   }
+  return returnValue;
 };
-const startServer = ({ cwd, command, env = {}, argv = [] }) =>
-  new Promise((resolve, reject) => {
-    let onTeardown;
-    const teardownPromise = new Promise((resolve) => (onTeardown = resolve));
+const startServer = ({ cwd, command, env = {}, argv = [] }) => {
+  const teardownPromise = new EProm();
+  const urlPromise = new EProm();
+  let stdout = '';
+  let stderr = '';
 
-    const hopsBin = resolveFrom(cwd, 'hops/bin');
-
-    const args = [hopsBin, command].concat(argv);
-    debug('Spawning server', [hopsBin, ...args].join(' '));
-    const started = spawn(process.argv[0], args, {
-      env,
-      cwd,
-    });
-    const teardown = () => {
-      started.kill();
-      return teardownPromise;
-    };
+  const hopsBin = resolveFrom(cwd, 'hops/bin');
+  const args = [hopsBin, command].concat(argv);
+  debug('Spawning server', args.join(' '), { env });
+  const started = spawn(process.argv[0], args, {
+    env,
+    cwd,
+  });
+  let stopServer = () => {
+    started.kill();
+    teardownPromise.resolve({ stderr, stdout });
+    return teardownPromise;
+  };
 
-    started.stdout.on('data', (data) => {
-      const line = data.toString('utf-8');
-      debug('stdout >', line);
-      const match = line.match(/listening at (.*)/i);
-      if (match) {
-        debug('found match:', match[1]);
-        const url = match[1];
-        resolve({ url, teardown });
-      }
-    });
+  started.stdout.on('data', (data) => {
+    const line = data.toString('utf-8');
+    debug('stdout >', line);
+    stdout += line;
 
-    let stderr = '';
-    started.stderr.on('data', (data) => {
-      const line = data.toString('utf-8');
-      debug('stderr >', line);
-      stderr += line;
-    });
+    const [, url] = line.match(/listening at (.*)/i) || [];
+    if (url) {
+      debug('found match:', url);
+      urlPromise.resolve(url);
+    }
+  });
 
-    started.on('close', (code) => {
-      debug('Server stopped. exitcode:', code);
-      onTeardown(stderr);
-      if (code) {
-        reject(stderr);
-      }
-    });
+  started.stderr.on('data', (data) => {
+    const line = data.toString('utf-8');
+    debug('stderr >', line);
+    stderr += line;
+  });
 
-    started.on('error', (error) => reject(error));
+  started.on('close', (code) => {
+    debug('Server stopped. exitcode:', code);
+    stopServer = () => {};
   });
 
+  started.on('error', (error) => teardownPromise.reject(error));
```

Comment:
There is one thing missing here. You should add this:

```js
process.on('exit', () => {
  stopServer();
});
```

It's not guaranteed by unix systems to stop spawned processes in all cases. Even though it is not a detached process. To be on the save side, we want to stop all spawned processes.